app/javascript/src/javascripts/post_flags.js

```diff
@@ -1,17 +1,176 @@
 const PostFlags = {};
 
+
 PostFlags.init = function () {
+  if (PostFlags._initialized) {
+    console.debug("PostFlags.init: already initialized");
+    return;
+  }
+  PostFlags._initialized = true;
+  console.debug("PostFlags.init: initializing");
   for (const container of $(".post-flag-note")) {
     if (container.clientHeight > 72) $(container).addClass("expandable");
   }
 
   $(".post-flag-note-header").on("click", (event) => {
     $(event.currentTarget).parents(".post-flag-note").toggleClass("expanded");
   });
+
+
+  const flagReasonLabels = document.querySelectorAll(".flag-reason-label");
+  const noteField = document.getElementById("flag_note_field");
+  // Use a live collection so newly-checked inputs are reflected
+  const radioButtons = document.getElementsByName("post_flag[reason_name]");
+  // Find the form by traversing up from the note field, or fallback to the first form in the flag-dialog-body
+  let form = null;
+  if (noteField) {
+    form = noteField.closest("form");
+  } else {
+    const dialogBody = document.querySelector(".flag-dialog-body");
+    if (dialogBody) {
+      form = dialogBody.querySelector("form");
+    }
+  }
+  let errorMessage = null;
+
+  // Parse dataset boolean values robustly. Treat presence or values like "", "true", "1", "yes" as true.
+  function parseDatasetBoolean (val) {
+    if (val === undefined || val === null) return false;
+    const v = String(val).trim().toLowerCase();
+    if (v === "" || v === "true" || v === "1" || v === "yes") return true;
+    return false;
+  }
```

Comment:
The config file is written in ruby, and only `true` and `false` are valid booleans. Empty string, `1`, and `yes` are therefore not candidates for values we would receive from there. For that reason I believe this function is unnecessary.

Reply:
We can also use the very convenient inbuilt [Boolean function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) instead:

```suggestion
  function parseDatasetBoolean (val) {
    return Boolean(val);
  }
```

But we should inline that.